### PR TITLE
SOLR-16050: Upgrade to errorprone 2.11.0

### PR DESCRIPTION
**Pull request by risdenk**
https://issues.apache.org/jira/browse/SOLR-16050

### CODE REVIEW

File: gradle/validation/error-prone.gradle [hunk 1]
```diff
@@ -64,6 +64,7 @@ allprojects { prj ->
             '-Xep:ComparableType:OFF', // SolrTestCaseJ4.Doc and Fld are messy
 
             '-Xep:AlmostJavadoc:OFF',
+            '-Xep:AlreadyChecked:OFF',
```
Comment (risdenk) [inline gradle/validation/error-prone.gradle:67]: This check makes logic way more confusing in the few cases it came up.

File: gradle/validation/error-prone.gradle [hunk 2]
```diff
@@ -104,9 +105,9 @@ allprojects { prj ->
             '-Xep:MissingSummary:OFF',
             '-Xep:MixedMutabilityReturnType:OFF',
             '-Xep:ModifiedButNotUsed:OFF',
-            '-Xep:MutableConstantField:OFF',
```
Comment (risdenk) [inline gradle/validation/error-prone.gradle:107]: No longer a valid check.

File: gradle/validation/error-prone.gradle [hunk 3]
```diff
@@ -104,9 +105,9 @@ allprojects { prj ->
...
             '-Xep:MutablePublicArray:OFF',
             '-Xep:NarrowingCompoundAssignment:OFF',
+            '-Xep:NegativeCharLiteral:OFF',
```
Comment (risdenk) [inline gradle/validation/error-prone.gradle:110]: Solr uses negative chars in a few places correctly and so this check wasn't worth it.

**Comment by madrob** [comment]
So did the new version of ErrorProne give us any _useful_ checks?

**Comment by risdenk** [comment]
well @madrob I should say no new checks that were worth anything that caused changes in the Solr codebase:

https://github.com/google/error-prone/releases/tag/v2.11.0

There are definitely new checks, but didn't require any changes to Solr.

**Review by madrob — APPROVED** [review]

**Pull request merged**